avm/utilities/pipelines/e2eValidation/regionSelector/Set-AzDeploymentRegion.ps1

```diff
@@ -0,0 +1,123 @@
+
+.PARAMETER usePairedRegionsOnly
+Optional. If set, only paired regions will be returned.
+
+.PARAMETER useKnownRegionsOnly
+Optional. If set, only known regions will be returned.
+
+.PARAMETER knownRegionList
+Optional. The list of known regions to use when useKnownRegionsOnly is set. ARM Deployment are limited to 10 active deployment regions per subscription.
+
+.EXAMPLE
+Set-AzDeploymentRegion -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US") -usePairedRegionsOnly -useKnownRegionsOnly
+
+Get the recommended regions using known regions excluding the regions specified in the excludedRegions parameter and only paired regions.
+
+.EXAMPLE
+Set-AzDeploymentRegion -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US") -usePairedRegionsOnly
+
+Get the recommended regions excluding the regions specified in the excludedRegions parameter and only paired regions.
+
+.EXAMPLE
+Set-AzDeploymentRegion -excludedRegions @("West Europe", "Brazil South", "West US 2", "East US 2", "South Central US")
+
+Get the recommended regions excluding the regions specified in the excludedRegions parameter.
+
+.EXAMPLE
+Set-AzDeploymentRegion -usePairedRegionsOnly
+
+Get the recommended paired regions.
+#>
+
+function Set-AzDeploymentRegion {
+  param (
+    [Parameter(Mandatory = $false)]
+    [array] $excludedRegions = @(
+      "westeurope",
+      "westus2",
+      "eastus2",
+      "centralus",
+      "southcentralus",
+      "brazilsouth",
+      "koreacentral",
+      "qatarcentral"
+    ),
+
+    [Parameter(Mandatory = $false)]
+    [cmdletbinding()]
+    [switch] $usePairedRegionsOnly,
+
+    [Parameter(Mandatory = $false)]
+    [cmdletbinding()]
+    [switch] $useKnownRegionsOnly,
+
+    [Parameter(Mandatory = $false)]
+    [array] $knownRegionList = @(
+      "northeurope",
+      "eastus",
+      "polandcentral",
+      "germanywestcentral",
+      "francecentral",
+      "westus3",
+      "australiaeast",
+      "uksouth"
+    )
+  )
+
+  $allRegions = Get-AzLocation -ExtendedLocation $true
+
+  # Filter regions where Location is in the known regions list
+  if ($useKnownRegionsOnly) {
+    $regionList = $allRegions | Where-Object { $_.Location -in $knownRegionList }
+  }
+  else {
+    $regionList = $allRegions
+  }
+
+  $exclusions = @()
+  if ($null -ne $excludedRegions) {
+    foreach ($region in $excludedRegions) {
+      if ($region -like "* *") {
+        $regionToExclude = $regionList | Where-Object { $_.DisplayName -eq $region } | Select-Object Location
+        $exclusions += $regionToExclude
+        Write-Verbose "Excluding region $regionToExclude"
+      }
+      else {
+        $regionToExclude = $regionList | Where-Object { $_.Location -eq $region } | Select-Object Location
+        $exclusions += $regionToExclude
+        Write-Verbose "Excluding region $regionToExclude"
+      }
+    }
+  }
```

Comment:
```suggestion
$exclusions = $excludedRegions | Where-Object { $_ -in ($_ -like '* *' ? $regionList.DisplayName : $regionList.Location) }
Write-Verbose ('Excluding regions: {0}' -f ($exclusions | ConvertTo-Json)) -Verbose
```

Note, `$excludedRegions` can never be `$null`

Reply:
Also note @ChrisSidebotham: If we apply this suggestion, then you won't need the `$exclusions.Location` reference later, but can just use `$exclusions`

Reply:
Example output:
```PowerShell
VERBOSE: Excluding regions: [
  "westeurope",
  "westus2",
  "eastus2",
  "centralus",
  "southcentralus",
  "brazilsouth",
  "koreacentral",
  "qatarcentral"
]
```

Reply:
@AlexanderSehr - I have tested this locally and $ecludions is coming up empty so I am not adding this in the current iteration

Reply:
Updated - Will only provide a list if it is being excluded from the `regionList` filtering